integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/projection/SearchProjectionIT.java

```diff
@@ -454,19 +459,20 @@ public void field_inNestedObject() {
 		StubMappingSearchTarget searchTarget = indexManager.createSearchTarget();
 
 		for ( FieldModel<?> fieldModel : indexMapping.nestedObject.supportedFieldModels ) {
-			SearchQuery<List<?>> query;
-			String fieldPath = indexMapping.nestedObject.relativeFieldName + "." + fieldModel.relativeFieldName;
-			Class<?> fieldType = fieldModel.type;
-
-			query = searchTarget.query()
-					.asProjections( searchTarget.projection().field( fieldPath, fieldType ).toProjection() )
-					.predicate( f -> f.matchAll().toPredicate() )
-					.build();
-			assertThat( query ).hasListHitsAnyOrder( b -> {
-				b.list( fieldModel.document1Value );
-				b.list( fieldModel.document2Value );
-				b.list( fieldModel.document3Value );
-				b.list( null ); // Empty document
+			SubTest.expectSuccess( fieldModel, model -> {
+				String fieldPath = indexMapping.nestedObject.relativeFieldName + "." + model.relativeFieldName;
+
+				assertThat(
+						searchTarget.query()
+								.asProjections( searchTarget.projection().field( fieldPath, model.type ).toProjection() )
```

Comment:
I think I would name it `asProjection()`. Otherwise, I would kinda expect the return type to be consistent when adding another projection to be returned.

The methods have totally different behavior.

Reply:
So, two things.

Regarding the ability to use this API with "generic" code: generic code will call the `asProjections()` method that accepts an array of projections as a parameter. It will read something like `.asProjections( myProjections.toArray( new SearchProjection[0] ) )`. So the fact that an overload with a single parameter exists does not matter: the method will return a `SearchQuery<List<?>>`, even if their array only contains one element.

About consistency, the other methods are named asObjects/asEntities/asReferences, depending on where you look. But they only return one element per hit. So the "s" here is about the fact there can be multiple hits, so multiple objects/entities/references. With that reasoning, we should keep `asProjections()`, with an "s", even when there is one projection: there will still be multiple hits, so multiple "projection results".

So if you want to remove the "s", I say okay, but let's remove it from the other methods too (except the one that take a projection array in parameter).